js/modules/k6/experimental/tracing/client.go

```diff
@@ -0,0 +1,243 @@
+	vu modules.VU
+
+	// opts holds the client's configuration options.
+	opts options
+
+	// propagator holds the client's trace propagator, used
+	// to produce trace context headers for each supported
+	// formats: w3c, b3, jaeger.
+	propagator Propagator
+
+	// requestFunc holds the http module's request function
+	// used to emit HTTP requests in k6 script. The client
+	// uses it under the hood to emit the requests it
+	// instruments.
+	requestFunc HTTPRequestFunc
+}
+
+type (
+	// HTTPRequestFunc is a type alias representing the prototype of
+	// k6's http module's request function
+	HTTPRequestFunc func(method string, url goja.Value, args ...goja.Value) (*httpmodule.Response, error)
+)
+
+// NewClient instantiates a new tracing Client
+func NewClient(vu modules.VU, opts options) *Client {
+	rt := vu.Runtime()
+
+	// Instantiate the http module in our runtime and get its default exported object
+	httpModuleObj, ok := httpmodule.New().NewModuleInstance(vu).Exports().Default.(*goja.Object)
+	if !ok {
+		common.Throw(rt, errors.New("failed to initialize tracing client, unable to load http module"))
+	}
+
+	// Get the http module's request function
+	httpModuleRequest := httpModuleObj.Get("request")
+
+	// Export the http module's request function goja.Callable as a Go function
+	var requestFunc HTTPRequestFunc
+	if err := rt.ExportTo(httpModuleRequest, &requestFunc); err != nil {
+		common.Throw(
+			rt,
+			fmt.Errorf("failed initializing tracing client, unable to export http.request method; reason: %w", err),
+		)
+	}
+
+	client := &Client{vu: vu, requestFunc: requestFunc}
+	if err := client.Configure(opts); err != nil {
+		common.Throw(
+			rt,
+			fmt.Errorf("failed initializing tracing client, invalid configuration; reason: %w", err),
+		)
+	}
+
+	return client
+}
+
+// Configure configures the tracing client with the given options.
+func (c *Client) Configure(opts options) error {
+	if err := opts.validate(); err != nil {
+		return fmt.Errorf("invalid options: %w", err)
+	}
+
+	switch opts.Propagator {
+	case "w3c":
+		c.propagator = &W3CPropagator{}
+	case "b3":
+		c.propagator = &B3Propagator{}
+	case "jaeger":
+		c.propagator = &JaegerPropagator{}
+	default:
+		return fmt.Errorf("unknown propagator: %s", opts.Propagator)
+	}
+
+	c.opts = opts
+
+	return nil
+}
+
+// Request instruments the http module's request function with tracing headers,
+// and ensures the trace_id is emitted as part of the output's data points metadata.
+func (c *Client) Request(method string, url goja.Value, args ...goja.Value) (*httpmodule.Response, error) {
+	rt := c.vu.Runtime()
+
+	// The http module's request function expects the first argument to be the
+	// request body. If no body is provided, we need to pass null to the function.
+	if len(args) == 0 {
+		args = []goja.Value{goja.Null()}
+	}
+
+	traceID := TraceID{
+		Prefix:            k6Prefix,
+		Code:              k6CloudCode,
+		UnixTimestampNano: uint64(time.Now().UnixNano()),
```

Comment:
Does it make sense to have it in unit64 if the stdlib uses int64?

Reply:
I had the same question, and it is indeed a requirement for our internal projects. See [discussion](https://github.com/grafana/k6/pull/2838#discussion_r1064394243) 👍🏻 🙇🏻

Reply:
I'm not getting what's the issue for them, there is an API in between from our data type and their. So why should it be an issue for them?

Reply:
Summoning @Blinkuu to answer those questions 🪄 🧙🏻 

PS: be mindful of not leaking private information here 🙏🏻

Reply:
Resolved internally. I'll be changing this field to an `int64` 🤝